kanidm_client/tests/default_entries.rs

```diff
@@ -0,0 +1,622 @@
+    attrs
+        .iter()
+        .map(|attr| {
+            println!("Writing to {}", attr);
+            is_attr_writable(&rsclient, id, attr).unwrap()
+        })
+        .for_each(|is_ok| assert!(is_ok == is_writeable));
+}
+
+fn test_modify_group(rsclient: &KanidmClient, group_names: &[&str], is_modificable: bool) -> () {
+    // need user test created to be added as test part
+    group_names.iter().for_each(|group| {
+        println!("Testing group: {}", group);
+        ["description", "name"].iter().for_each(|attr| {
+            assert!(is_attr_writable(&rsclient, group, attr).unwrap() == is_modificable)
+        });
+        assert!(rsclient.idm_group_add_members(group, vec!["test"]).is_ok() == is_modificable);
+    });
+}
+
+// Users
+// - Read to all self attributes (within security constraints).
+// - Write to a limited set of self attributes, such as:
+//     name, displayname, legalname, ssh-keys, credentials etc.
+#[test]
+fn test_default_entries_rbac_users() {
+    run_test(|mut rsclient: KanidmClient| {
+        rsclient
+            .auth_simple_password("admin", ADMIN_TEST_PASSWORD)
+            .unwrap();
+
+        create_user_with_all_attrs(&mut rsclient, "self_account", Some("self_group"));
+        create_user_with_all_attrs(&mut rsclient, "other_account", Some("other_group"));
+
+        login_account(&mut rsclient, "self_account");
+
+        test_read_attrs(&rsclient, "self_account", &USER_READABLE_ATTRS, true);
+        test_read_attrs(&rsclient, "other_account", &USER_READABLE_ATTRS, true);
+
+        static GROUP_READABLE_ATTRS: [&str; 5] = ["class", "name", "spn", "uuid", "member"];
+        test_read_attrs(&rsclient, "self_group", &GROUP_READABLE_ATTRS, true);
+        test_read_attrs(&rsclient, "other_group", &GROUP_READABLE_ATTRS, true);
+
+        static USER_SENSITIVE_ATTRS: [&str; 2] = ["legalname", "mail"];
+        test_read_attrs(&rsclient, "other_account", &USER_SENSITIVE_ATTRS, false);
+
+        static SELF_READABLE_ATTRS: [&str; 1] = ["radius_secret"];
+        test_read_attrs(&rsclient, "self_account", &SELF_READABLE_ATTRS, true);
+        test_read_attrs(&rsclient, "other_account", &SELF_READABLE_ATTRS, false);
+
+        test_write_attrs(&rsclient, "self_account", &SELF_WRITEABLE_ATTRS, true);
+        test_write_attrs(&rsclient, "other_account", &SELF_WRITEABLE_ATTRS, false);
+
+        static NON_SELF_WRITEABLE_ATTRS: [&str; 5] =
+            ["spn", "class", "memberof", "gidnumber", "uuid"];
+        test_write_attrs(&rsclient, "self_account", &NON_SELF_WRITEABLE_ATTRS, false);
+    });
+}
+
+// Account Managers
+// read and write to accounts, including write credentials but NOT private data (see people manager)
+// ability to lock and unlock accounts, excluding high access members.
+#[test]
+fn test_default_entries_rbac_account_managers() {
+    run_test(|mut rsclient: KanidmClient| {
+        rsclient
+            .auth_simple_password("admin", ADMIN_TEST_PASSWORD)
```

Comment:
Gah I just realised. You don't want to do this as admin. You need to bootstrap a test-admin that you add to each priv group one at a time, so that you guarantee you are only testing that one expect groups privileges rather than admin who comes with a set of rights already.

Reply:
mmmm I use admin just too bootstrap my test environment. In that case I create a account_manager and a test entry to check all behaviours against them.

Reply:
This is not clear to me, what could we do here?

Reply:
No, we can still use the admin for bootstrapping stuff, just the test needs to be checked as account_manager which I see you are doing now, so I think it's okay.